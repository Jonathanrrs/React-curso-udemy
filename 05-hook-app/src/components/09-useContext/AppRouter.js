import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from 'react-router-dom';
import { AboutScren } from './AboutScren';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

/* exact es para que sea exacto el nombre de las rutas */

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route exact path="/" component={HomeScreen}/>
                    <Route exact path="/about" component={AboutScren}/>
                    <Route exact path="/login" component={LoginScreen}/>
                    {/* <Route component={HomeScreen}/> no importa el path, te manda al home */}
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}
