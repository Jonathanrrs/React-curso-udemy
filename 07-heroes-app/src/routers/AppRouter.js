/* Es el principal router de la app */

import React, { useContext } from 'react';
import {
  BrowserRouter as Router,
  Switch,
} from "react-router-dom";

import { AuthContext } from '../auth/AuthContext';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';


export const AppRouter = () => {

  const { user } = useContext(AuthContext);

  return (
    <Router>
      <div>
        {/* esto lo  hacemos para que el login no tenga el estilo del otro router */}
        <Switch>
          <PublicRoute exact path="/login" component={LoginScreen} isAuthenticated={user.logged} />
          {/* no debe ser exact porque sino no funciona */}
          {/* aqui ponemos nuestro funcional component para proteger las rutas en caso de no logeado */}
          <PrivateRoute path="/" component={DashboardRoutes} isAuthenticated={user.logged} />
        </Switch>
      </div>
    </Router>
  )
}
