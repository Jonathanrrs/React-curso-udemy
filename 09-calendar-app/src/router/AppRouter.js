import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { LoginScreen } from '../componentes/auth/LoginScreen';
import { CalendarScreen } from '../componentes/calendar/CalendarScreen';

export const AppRouter = () => {
    return (
        <>
            <Router>
                <div>
                    <Switch>
                        <Route path="/login" exact component={LoginScreen}></Route>
                        <Route path="/" exact component={CalendarScreen}></Route>
                        <Redirect to="/" />
                    </Switch>
                </div>
            </Router>
        </>

    )
}
