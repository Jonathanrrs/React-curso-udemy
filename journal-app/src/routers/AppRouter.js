import React, { useEffect, useState } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Redirect
} from 'react-router-dom';
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { firebase } from '../firebase/firebase-config';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';
import { loadNotes } from '../helpers/loadNotes';
import { setNotes } from '../actions/notes';


export const AppRouter = () => {

    const dispatch = useDispatch();

    const [checking, setChecking] = useState(true);

    const [isLoggedIn, setIsLoggedIn] = useState(false);



    /* mantener el state si se recarga la pÃ¡gina */
    useEffect(() => {
        firebase.auth().onAuthStateChanged(async(user) => {

            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);

                const notes = await loadNotes(user.uid);
                dispatch(setNotes(notes));
            } else {
                setIsLoggedIn(false)
            }
            /* si tenemos el usuario mostramos la vista */
            setChecking(false);

        });


    }, [dispatch, setChecking, setIsLoggedIn])

    if (checking) {
        return (
            <h1>Wait...</h1>
        )
    }

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute
                        path="/auth"
                        component={AuthRouter}
                        isAuthenticated={isLoggedIn}
                    >
                    </PublicRoute>

                    <PrivateRoute
                        exact
                        path="/"
                        component={JournalScreen}
                        isAuthenticated={isLoggedIn}
                    >
                    </PrivateRoute>
                    <Redirect to="/auth/login" />
                </Switch>
            </div>
        </Router>
    )
}
