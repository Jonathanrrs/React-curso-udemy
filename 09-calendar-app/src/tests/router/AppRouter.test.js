import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { AppRouter } from '../../router/AppRouter';




const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        checking: true
    }
};
const store = mockStore(initState);
// store.dispatch = jest.fn();



describe('Pruebas en AppRouter', () => {
   test('Debe mostrar el espere... ', () => {

    const wrapper = mount(
        <Provider store={store}>
            <AppRouter />
        </Provider>
    );
    expect(wrapper).toMatchSnapshot();
});

    
});
