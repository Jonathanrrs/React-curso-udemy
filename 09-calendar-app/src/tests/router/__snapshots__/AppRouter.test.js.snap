// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe mostrar el espere...  1`] = `
<h5>
  Espere...
</h5>
`;
